server/src/main/java/org/elasticsearch/search/query/QueryPhase.java

```diff
@@ -202,36 +201,28 @@ static void addCollectorsAndSearch(SearchContext searchContext) throws QueryPhas
                 searcher.addQueryCancellation(timeoutRunnable);
             }
 
-            try {
-                QueryPhaseResult queryPhaseResult = searcher.search(query, collectorManager);
-                if (searchContext.getProfilers() != null) {
-                    searchContext.getProfilers().getCurrentQueryProfiler().setCollectorResult(queryPhaseResult.collectorResult());
-                }
-                queryResult.topDocs(queryPhaseResult.topDocsAndMaxScore(), queryPhaseResult.sortValueFormats());
-                if (searcher.timeExceeded()) {
-                    assert timeoutRunnable != null : "TimeExceededException thrown even though timeout wasn't set";
-                    if (searchContext.request().allowPartialSearchResults() == false) {
-                        throw new QueryPhaseExecutionException(searchContext.shardTarget(), "Time exceeded");
-                    }
-                    queryResult.searchTimedOut(true);
-                }
-                if (searchContext.terminateAfter() != SearchContext.DEFAULT_TERMINATE_AFTER) {
-                    queryResult.terminatedEarly(queryPhaseResult.terminatedAfter());
-                }
-                ExecutorService executor = searchContext.indexShard().getThreadPool().executor(ThreadPool.Names.SEARCH);
-                assert executor instanceof TaskExecutionTimeTrackingEsThreadPoolExecutor
-                    || (executor instanceof EsThreadPoolExecutor == false /* in case thread pool is mocked out in tests */)
-                    : "SEARCH threadpool should have an executor that exposes EWMA metrics, but is of type " + executor.getClass();
-                if (executor instanceof TaskExecutionTimeTrackingEsThreadPoolExecutor rExecutor) {
-                    queryResult.nodeQueueSize(rExecutor.getCurrentQueueSize());
-                    queryResult.serviceTimeEWMA((long) rExecutor.getTaskExecutionEWMA());
-                }
-            } finally {
-                // Search phase has finished, no longer need to check for timeout
-                // otherwise aggregation phase might get cancelled.
-                if (timeoutRunnable != null) {
-                    searcher.removeQueryCancellation(timeoutRunnable);
```

Comment:
I think I lost track of where we are not removing the query cancellation runnable? Oh perhaps this is no longer needed while before we'd need to remove it before executing the aggregation phase?

Reply:
Because we are not running aggregations after the search any longer. No needed it any longer.